skyportal/facility_apis/atlas.py

```diff
@@ -0,0 +1,310 @@
+            DBSession()
+            .query(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .one()
+        )
+
+        instrument = (
+            Instrument.query_records_accessible_by(request.requester)
+            .join(Allocation)
+            .join(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        content = req.transactions[0].response["content"]
+        content = json.loads(content)
+
+        r = requests.get(
+            content["url"],
+            headers={
+                'Authorization': f"Token {altdata['api_token']}",
+                'Accept': 'application/json',
+            },
+        )
+        r.raise_for_status()
+
+        if r.status_code == 200:
+            if r.json()['finishtimestamp']:
+                result_url = r.json()['result_url']
+                request.status = (
+                    f"Task is complete with results available at {result_url}"
+                )
+
+                s = requests.get(
+                    result_url,
+                    headers={
+                        'Authorization': f"Token {altdata['api_token']}",
+                        'Accept': 'application/json',
+                    },
+                )
+                s.raise_for_status()
+
+                # ATLAS response looks like
+                """
+###MJD          m      dm   uJy   duJy F err chi/N     RA       Dec        x        y     maj  min   phi  apfit mag5sig Sky   Obs
+59226.235875  16.177  0.012  1228   15 c  0  54.64 342.45960  51.26340  7768.79  7767.00 2.53 2.39 -63.4 -0.375 19.58 21.54 01a59226o0051c
+59228.242600  16.258  0.017  1140   20 c  0   7.87 342.45960  51.26340  2179.59  9252.78 3.41 3.09 -51.0 -0.396 19.28 21.28 02a59228o0102c
+59228.246262  16.582  0.021   846   18 c  0  28.37 342.45960  51.26340  2162.23  9213.32 3.53 3.25 -52.3 -0.366 19.14 21.26 02a59228o0110c
+59228.252679  16.451  0.019   954   18 c  0  13.76 342.45960  51.26340  2218.02  9291.76 3.34 3.03 -49.8 -0.389 19.17 21.24 02a59228o0124c
+59228.265532  17.223  0.049   469   23 c  0   3.90 342.45960  51.26340  2237.25  9167.94 4.31 3.88 -43.7 -0.473 18.95 21.20 02a59228o0152c
+                """
+
+                try:
+                    df = pd.read_csv(
+                        StringIO(s.text.replace("###", "")), delim_whitespace=True
+                    )
+                except Exception as e:
+                    raise ValueError(f'Format of response not understood: {e.message}')
+
+                desired_columns = ['MJD', 'RA', 'Dec', 'm', 'dm', 'mag5sig', 'F']
+                for col in desired_columns:
+                    if col not in list(set(df.columns.values)):
+                        raise ValueError(f'Missing expected column: {col}')
+
+                df.rename(
```

Comment:
If you are going to do a rename anyway, then you may just as well do the rename earlier, replace `###MJD` with `mjd`, and not perform the stringio replacement.

Reply:
Added.